core/tests/test_utils.py

```diff
@@ -1685,6 +1651,30 @@ def swap(self, obj, attr, newvalue):
         finally:
             setattr(obj, attr, original)
 
+    @contextlib.contextmanager
+    def login_context(self, email, is_super_admin=False):
```

Comment:
Does this behave correctly if the email passed is invalid (doesn't correspond to a user)?

Reply:
Hmm, it will do whatever `self.login` does, so I guess so? Not sure what you mean by 'behave correctly'.